modyn/metadata_database/models/selector_state_metadata.py

```diff
@@ -45,7 +47,7 @@ class SelectorStateMetadata(
     __tablename__ = "selector_state_metadata"
 
     @staticmethod
-    def add_pipeline(pipeline_id: int, session: Session, engine: Engine) -> PartitionByMeta:
+    def add_pipeline(pipeline_id: int, session: Session, engine: Engine) -> PartitionByMeta | None:
```

Comment:
I think this should rather be Optional[PartitionByMeta] for consistency

Reply:
I will add `pyupgrade` soonish. That will upgrade all old python syntax to Python 3.11+ and eliminate the inconsistencies.